modules/order/src/EventSubscriber/OrderMailer.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Drupal\commerce_order\EventSubscriber;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Mail\MailManagerInterface;
+use Drupal\Core\Render\Markup;
+use Drupal\Core\Render\RenderContext;
+use Drupal\Core\Render\Renderer;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\state_machine\Event\WorkflowTransitionEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Listens to order transition events and sends emails.
+ */
+class OrderMailer implements EventSubscriberInterface {
+  use StringTranslationTrait;
+
+  /**
+   * The order type entity storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $orderTypeStorage;
+
+  /**
+   * The entity view builder for orders.
+   *
+   * @var \Drupal\Core\Entity\EntityViewBuilderInterface
+   */
+  protected $orderViewBuilder;
+
+  /**
+   * The mail manager service.
+   *
+   * @var \Drupal\Core\Mail\MailManagerInterface
+   */
+  protected $mailManager;
+
+  /**
+   * The Drupal renderer.
+   *
+   * @var \Drupal\Core\Render\RendererInterface
+   */
+  protected $renderer;
+
+  /**
+   * Constructs a new OrderMailer object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Mail\MailManagerInterface $mail_manager
+   *   A mail manager service.
+   * @param \Drupal\Core\Render\Renderer $renderer
+   *   The Drupal renderer.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, MailManagerInterface $mail_manager, Renderer $renderer) {
+    $this->orderTypeStorage = $entity_type_manager->getStorage('commerce_order_type');
+    $this->orderViewBuilder = $entity_type_manager->getViewBuilder('commerce_order');
+    $this->mailManager = $mail_manager;
+    $this->renderer = $renderer;
+  }
+
+  /**
+   * Sends order receipt to customer and store email.
+   *
+   * @param \Drupal\state_machine\Event\WorkflowTransitionEvent $event
+   *   The event we subscribed to.
+   */
+  public function customerOrderReceipt(WorkflowTransitionEvent $event) {
+    /** @var \Drupal\commerce_order\Entity\OrderInterface $order */
+    $order = $event->getEntity();
+
+    // Check if the emails are enabled for this order type.
+    /** @var \Drupal\commerce_order\Entity\OrderTypeInterface $order_type */
+    $order_type = $this->orderTypeStorage->load($order->bundle());
+    if (!$order_type->shouldSendReceipt()) {
+      return;
+    }
+
+    $customer = $order->getCustomer();
+    $store = $order->getStore();
+
+    // Build the body of the email.
+    $build = [
+      '#theme' => 'commerce_order_receipt',
+      '#order' => $order,
+      '#additional_info' => [
```

Comment:
Key name does not match template variable (additional_info instead of additional_information).
It doesn't feel like we should have this variable, I think this text should move to the template, people will override the template for customizations anyway, giving them a preprocess shortcut for customizing the footer text doesn't really help (and if it's another module, they'd probably want more control than dumping it all in the footer)